subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -575,6 +575,54 @@ static int engine_remove_observer(const uint8_t *token, uint8_t tkl)
 	return 0;
 }
 
+#if defined(CONFIG_LOG)
+char *lwm2m_path_log_strdup(struct lwm2m_obj_path *path)
+{
+	char buf[sizeof("65535/65535/65535/65535")];
+	size_t cur = sprintf(buf, "%u", path->obj_id);
+
+	if (path->level > 1) {
+		cur += sprintf(buf, "/%u", path->obj_inst_id);
+	}
+	if (path->level > 2) {
+		cur += sprintf(buf, "/%u", path->res_id);
+	}
+	if (path->level > 3) {
+		cur += sprintf(buf, "/%u", path->res_inst_id);
+	}
+
+	return log_strdup(buf);
+}
+#endif /* CONFIG_LOG */
+
+#if defined(CONFIG_LWM2M_CANCEL_OBSERVE_BY_PATH)
+static int engine_remove_observer_by_path(struct lwm2m_obj_path *path)
+{
+	struct observe_node *obs, *found_obj = NULL;
+	sys_snode_t *prev_node = NULL;
+
+	/* find the node index */
+	SYS_SLIST_FOR_EACH_CONTAINER(&engine_observer_list, obs, node) {
+		if (memcmp(path, &obs->path, sizeof(*path)) == 0) {
+			found_obj = obs;
+			break;
+		}
+
+		prev_node = &obs->node;
+	}
+
+	if (!found_obj) {
+		return -ENOENT;
+	}
+
+	LOG_INF("Removing observer for path %s", path);
```

Comment:
`lwm2m_path_log_strdup(path)`?

Reply:
Yeah I fixed it but force pushed to the wrong branch :D